pkg/cluster/versions.go

```diff
@@ -42,12 +42,12 @@ func EnsureOpenshiftVPrefix(v string) string {
 // GetEnabledVersions returns the versions with enabled=true, and the one that has default=true.
 // The returned strings are the IDs without "openshift-v" prefix (e.g. "4.6.0-rc.4-candidate")
 // sorted in approximate SemVer order (handling of text parts is somewhat arbitrary).
-func GetEnabledVersions(client *cmv1.Client, channelGroup string) (
+func GetEnabledVersions(client *cmv1.Client, channelGroup string, gcpMarketplaceEnabled string) (
 	versions []string, defaultVersion string, err error) {
 	collection := client.Versions()
 	page := 1
 	size := 100
-	filter := "enabled = 'true'"
+	filter := fmt.Sprintf("enabled = 'true' AND gcp_marketplace_enabled = '%s'", gcpMarketplaceEnabled)
 	if channelGroup != "" {
 		filter = fmt.Sprintf("%s AND channel_group = '%s'", filter, channelGroup)
 	}
```

Comment:
it would be probably safer to only set the gcp_marketplace_enabled restriction if a value is explicitly  provided (like channel group).

I would keep `filter := "enabled = 'true'"` as the base query only.

Reply:
Changed